src/replication.c

```diff
@@ -1777,6 +1778,11 @@ void readSyncBulkPayload(connection *conn) {
             if (server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB) {
                 /* Discard potentially partially loaded tempDb. */
                 disklessLoadDiscardTempDb(diskless_load_tempDb);
+                serverLog(LL_NOTICE, "MASTER <-> REPLICA sync: Discarded temporary DB");
+
+                moduleFireServerEvent(REDISMODULE_EVENT_REPL_ASYNC,
+                                      REDISMODULE_SUBEVENT_REPL_ASYNC_ABORTED,
+                                      NULL);
```

Comment:
i think that maybe we better fire the event before discarding the db.
maybe modules will find that more useful.
i.e. they're gonna get a bunch of `free` callbacks and they'll wanna know if these callback are for actual data or temp data.
@MeirShpilraien do you think we need to have a post discard event too?

Reply:
Should it then be:
1. REDISMODULE_SUBEVENT_REPL_ASYNC_ABORTING
2. REDISMODULE_SUBEVENT_REPL_ASYNC_ABORTED
?

Reply:
i'm ok with what you suggested, but maybe a better alternative could be something like:
1. REDISMODULE_SUBEVENT_REPL_ASYNC_DISCARD_START
2. REDISMODULE_SUBEVENT_REPL_ASYNC_DISCARD_DONE

(or ABORT_START / ABORT_DONE).
the advantage is that it's a common prefix for a start and end of the same operation (rather than two similar words with ING and ED suffix)

Reply:
ohh, i just realized the discard is (always) async. so the module can't use these START / DONE notifications to know know if a free callback is part of the discarded database or the other one...
i suppose that can be a real issue for some modules, and they'll have no choice but avoid declaring this capability.
for the other ones, i suppose firing the event before we start discarding is slightly better.
@MeirShpilraien what do you think?

Reply:
@oranagra agree, do not see other choose ...

Reply:
@oranagra @MeirShpilraien, maybe a small confusion, but what we have added is REDISMODULE_SUBEVENT_REPL_ASYNC_LOAD_ABORTED, for the case we started loading on tempDb then discarded due to failure.
There's also the discard that happens after successful load. In this case we fire a REDISMODULE_SUBEVENT_REPL_ASYNC_LOAD_COMPLETED

Should the ABORT really become DISCARD (from your comment here) and be fired only on the failure scenario?

I guess also the REDISMODULE_SUBEVENT_REPL_ASYNC_LOAD_COMPLETED should be fired before call to discard tempDb on successful case (instead of before as it is now)?

Reply:
@eduardobr please disregard our recent comments about the before and after.
i.e. all the ABORTING / ABORTED and DISCARD_START / DISCARD_DONE.
they where all about a way for the module to be able to distinguish between `free` callbacks of the main db, and the temp db, but since the freeing is always done in the background, the before and after notifications are no good.

the only think that's left to change IMHO is that we prefer that the ABORTED and COMPLETED events be fired before the freeing starts rather than after it.

Reply:
Done then ;)
Thanks